File: typescript/api/src/rooms/startRoom.ts
import {BoxWithBomb} from "../items/boxWithBomb";
import {BoxWithGoldenKey} from "../items/boxWithGoldenKey";
import {DoorToFinalRoom} from "../items/doorToFinalRoom";
import {Table} from "../items/table";
import {Room} from "../base/room";

export class StartRoom extends Room {

    public constructor() {
        super("start_room");
        this.objects.push(new Table("table"));
        this.objects.push(new BoxWithBomb("box_with_bomb"));
        this.objects.push(new BoxWithGoldenKey("box_with_golden_key"));
        this.objects.push(new DoorToFinalRoom("door_to_final_room"));
    }

    public getShortDescription(): string {
        return "Starting room";
    }

    public getFullDescription(): string {
        return "A very strange room. There is a table, two boxes and a closed door";
    }

}

File: java/api/src/main/java/game/engine/java/api/rooms/StartRoom.java
package game.engine.java.api.rooms;

import game.engine.java.api.base.Room;
import game.engine.java.api.items.Chest;
import game.engine.java.api.items.Door;
import game.engine.java.api.items.Table;

public class StartRoom extends Room {

    public StartRoom() {
        super("start_room");
        // TODO add more objects -----
        super.getObjects().add(new Door("door"));
        super.getObjects().add(new Chest("chest"));
        super.getObjects().add(new Table("table"));
    }

    public String getShortDescription() {
        return "Starting room";
    }

    public String getFullDescription() {
        return "A very strange room. There is a table, two boxes and a closed door";
    }
}
